### PR TITLE
Clean up headers

### PR DESCRIPTION
**Pull request by wingdeans**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

* Remove duplicate declarations
* Remove undefined functions
* Added necessary `#include`s so all headers are standalone

A list of undefined functions was made by diffing the exported elf symbols and header definitions.
The list was manually verified by grepping.

Only a few functions did have definitions:
`rz_range_merge` : `#if 0`
`rz_core_echo` : `external` (not sure what to do)
`rz_magic_load_buffer` : defined but not present?

Some files may need additional consideration:
`rz_util/rz_pkcs.h` and `rz_util/rz_x509.h` relied on external declarations and were not scanned
`rz_pool.h` might be removable since all its functions are undefined
~~rz_heap_jemalloc.h doesn't define any functions and #includes a nonexistent jemalloc_internal.h file~~

Some functions may have definitions but are not exposed with the same name as the header declaration.
Rather than removing the header declaration, these functions should be exposed
...

**Test plan**

This should have only affected undefined and duplicate functions.

### CODE REVIEW

File: librz/include/rz_types_base.h [hunk 1]
```diff
@@ -97,7 +97,6 @@ typedef struct _utX {
 #define ST32_MAX  0x7FFFFFFF
 #define ST32_MIN  (-ST32_MAX - 1)
 #define UT32_MAX  0xFFFFFFFFU
-#define UT32_MIN  0U
```
Comment (wargio) [inline librz/include/rz_types_base.h:100]: keep this

File: librz/include/rz_analysis.h [hunk 2]
```diff
@@ -68,8 +68,6 @@ typedef struct rz_analysis_range_t {
 	RBNode rb;
 } RzAnalysisRange;
 
-#define RZ_ANALYSIS_GET_OFFSET(x, y, z) \
-	(x && x->binb.bin && x->binb.get_offset) ? x->binb.get_offset(x->binb.bin, y, z) : -1
```
Comment (wargio) [inline librz/include/rz_analysis.h:72]: This is used in analysis_dalvik.c

File: librz/include/rz_core.h [hunk 3]
```diff
@@ -486,9 +483,7 @@ RZ_API RzCmdStatus rz_core_cmd_rzshell(RzCore *core, const char *cmd, int log);
 RZ_API char *rz_core_editor(const RzCore *core, const char *file, const char *str);
 RZ_API int rz_core_fgets(char *buf, int len, void *user);
 RZ_API RzFlagItem *rz_core_flag_get_by_spaces(RzFlag *f, ut64 off);
-RZ_API int rz_core_cmdf(RzCore *core, const char *fmt, ...) RZ_PRINTF_CHECK(2, 3);
 RZ_API int rz_core_flush(RzCore *core, const char *cmd);
-RZ_API int rz_core_cmd0(RzCore *core, const char *cmd);
```
Comment (wargio) [inline librz/include/rz_core.h:491]: for the others, these are repeated below

File: librz/include/rz_egg.h [hunk 4]
```diff
@@ -209,7 +209,6 @@ RZ_API int rz_egg_padding(RzEgg *egg, const char *pad);
 RZ_API bool rz_egg_assemble(RzEgg *egg);
 RZ_API bool rz_egg_assemble_asm(RzEgg *egg, char **asm_list);
 RZ_API void rz_egg_pattern(RzEgg *egg, int size);
-RZ_API RzBuffer *rz_egg_get_bin(RzEgg *egg);
```
Comment (wargio) [inline librz/include/rz_egg.h:212]: dup again

File: librz/include/rz_socket.h [hunk 5]
```diff
@@ -263,15 +263,6 @@ RZ_API bool rz_run_parsefile(RzRunProfile *p, const char *b);
 RZ_API char *rz_run_get_environ_profile(char **env);
 
 /* rapipe */
```
Comment (wargio) [inline librz/include/rz_socket.h]: delete this comment too

**Comment by wargio** [comment]
- `rz_pool.h` can be deleted.
- what declaration is missing in `rz_util/rz_pkcs.h` and `rz_util/rz_x509.h` ?
- `rz_magic_load_buffer` is defined in `librz/magic/magic.c`
- `rz_core_echo` needs to go, remove that and the line with the `external` declaration.

File: librz/include/rz_types_base.h (hunk 1 above)
Comment (wingdeans) [reply librz/include/rz_types_base.h:100]: duplicate definition on line 93

File: librz/include/rz_analysis.h (hunk 2 above)
Comment (wingdeans) [reply librz/include/rz_analysis.h:72]: duplicate definition on line 105

**Comment by wingdeans** [comment]
rz_pkcs and rz_x509 are fine, I just needed to include rz_asn1 and rz_pj

rz_magic_load_buffer wasn't defined for me because I had USE_LIB_MAGIC.
I've defined it based off of the non-system implementation. Let me know if it needs changes or should be removed

**Comment by wargio** [comment]
Ok, add the missing headers then, for x509/pkcs7
Regarding USE_LIB_MAGIC, add that to the header, i guess like `#if USE_LIB_MAGIC ... #endif`

**Review by ret2libc — APPROVED** [review]
Nicely done!

**Review by XVilka — APPROVED** [review]

File: librz/magic/magic.c [hunk 6]
```diff
@@ -48,6 +48,14 @@ RZ_API const char *rz_magic_error(RzMagic *m) {
 RZ_API void rz_magic_setflags(RzMagic *m, int f) {
 	magic_setflags(m, f);
 }
+RZ_API bool rz_magic_load_buffer(RzMagic *m, const char *f) {
+	if (*magicdata == '#') {
```
Comment (wingdeans) [inline librz/magic/magic.c:52]: Should be f.
The tests are passing but the build will break with USE_LIB_MAGIC

Comment (XVilka) [reply librz/magic/magic.c:52]: @wingdeans  please send a PR

**Pull request merged**